spec/router_spec.lua

```diff
@@ -41,12 +41,6 @@ describe("Router", function()
         assert.same(dummy.params, {id="2"})
       end)
 
-      it("supports an extension on a param", function()
```

Comment:
I think adding support for dot (.) conflict with your requirement of supporting extension for the resource.

e.g  you have a test case  ```/resource/1.json``` where you expect resource value to be `1 `
 By adding support for ., it will return `1.json`
e.g `/users/rohit.joshi@github.com` should return `rohit.joshi@github.com`
So unless we remove support for extension, dot support will not work.

Regarding defining constants,  I had ran this test long ago and most of the time JIT does optimize but with millions of requests, I was able to see the difference. I do agree performance gain may not be significant compared to readability.